service/common/src/main/java/org/apache/polaris/service/context/RealmContextConfiguration.java

```diff
@@ -33,6 +33,16 @@ public interface RealmContextConfiguration {
   /** The header name that contains the realm identifier. */
   String headerName();
 
+  /**
+   * Whether to require the realm header to be present in the request. If this is true and the realm
+   * header is not present, the request will be rejected. If this is false and the realm header is
+   * not present, the default realm will be used.
+   *
+   * <p>Note: this is actually only enforced when using {@link DefaultRealmIdResolver}. When using
+   * {@link TestRealmIdResolver}, this setting is ignored.
+   */
+  boolean requireHeader();
```

Comment:
Should this be true by default?

Reply:
Defaulting to `true` would be a behaviour change at the REST/HTTP level. I'd prefer to keep old behaviour in this PR and change defaults later.

Reply:
WFM

Reply:
Yeah changing this to true would break downstreams. But we can mention this in the "Configuring for production" docs.